kafka/integration_test.go

```diff
@@ -174,39 +178,63 @@ func handleTestEvent(c *Consumer, mt *msgtracker, expCnt int, ev Event) bool {
 		if e.TopicPartition.Error != nil {
 			mt.t.Errorf("Error: %v", e.TopicPartition)
 		}
-		mt.msgs[mt.msgcnt] = e
-		mt.msgcnt++
-		if mt.msgcnt >= int64(expCnt) {
-			return false
+		if mt.msgcnt == int64(cap(mt.msgs)) {
+			mt.t.Logf("***** Buffer full. not saving message: %v\n", e)
+		} else {
+			// mt.t.Logf("Saving message to mt.msgs[%d]: %v\n", mt.msgcnt, e)
+			mt.msgs[mt.msgcnt] = e
+			mt.msgcnt++
 		}
 	case PartitionEOF:
 		break // silence
+	case *Stats:
+		mt.t.Logf("Stats: %v", e)
```

Comment:
The JSON should be verified to be syntactically correct, if there is an easy way to do that with Go without requiring full-blow corresponding go types